resources/bigtop-1.1.0/BIGTOP-2467.patch

```diff
@@ -43,6 +43,71 @@ index 69d98c4..b646e60 100644
  # purge snapshots every day
  autopurge.purgeInterval=24
--- 
-2.7.4
 
+From 0560313f10aac5ce7ea47e919edda03ea7a05319 Mon Sep 17 00:00:00 2001
+From: Pete Vander Giessen <petevg@gmail.com>
+Date: Tue, 21 Jun 2016 10:13:22 -0400
+Subject: [PATCH 2/3] Fixed some references to ensemble that I had missed on
+ the first pass.
+
+---
+ bigtop-deploy/puppet/modules/hadoop_zookeeper/manifests/init.pp | 2 +-
+ bigtop-deploy/puppet/modules/hadoop_zookeeper/tests/init.pp     | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/bigtop-deploy/puppet/modules/hadoop_zookeeper/manifests/init.pp b/bigtop-deploy/puppet/modules/hadoop_zookeeper/manifests/init.pp
+index 6965090..31a609e 100644
+--- a/bigtop-deploy/puppet/modules/hadoop_zookeeper/manifests/init.pp
++++ b/bigtop-deploy/puppet/modules/hadoop_zookeeper/manifests/init.pp
+@@ -61,7 +61,7 @@
+   class server($myid,
+                 $port = "2181",
+                 $datadir = "/var/lib/zookeeper",
+-                $ensemble = ["localhost:2888:3888"],
++                $ensemble = [$myid, "localhost:2888:3888"],
+                 $kerberos_realm = $hadoop_zookeeper::kerberos_realm,
+   ) inherits hadoop_zookeeper {
+     include common
+diff --git a/bigtop-deploy/puppet/modules/hadoop_zookeeper/tests/init.pp b/bigtop-deploy/puppet/modules/hadoop_zookeeper/tests/init.pp
+index 0340f10..7158714 100644
+--- a/bigtop-deploy/puppet/modules/hadoop_zookeeper/tests/init.pp
++++ b/bigtop-deploy/puppet/modules/hadoop_zookeeper/tests/init.pp
+@@ -16,5 +16,5 @@
+ hadoop_zookeeper::client { "zoo visitor": }
+ hadoop_zookeeper::server { "test-oozie-server":
+   myid => "0",
+-  ensemble => ["foo:2888:3888", "bar:2888:3888", "baz:2888:3888"],
++  ensemble => [[0, "foo:2888:3888"], [1, "bar:2888:3888"], [2, "baz:2888:3888"]],
+ }
+
+From 05671ac235319c3993ae92c50900dc7064d8efaf Mon Sep 17 00:00:00 2001
+From: Pete Vander Giessen <petevg@gmail.com>
+Date: Wed, 3 Aug 2016 17:27:47 -0400
+Subject: [PATCH 3/3] Added backwards compatibilty for old way of doing
+ ensemble.
+
+Should prevent us from breaking existing puppet scripts.
+---
+ bigtop-deploy/puppet/modules/hadoop_zookeeper/templates/zoo.cfg | 8 +++++++-
+ 1 file changed, 7 insertions(+), 1 deletion(-)
+
+diff --git a/bigtop-deploy/puppet/modules/hadoop_zookeeper/templates/zoo.cfg b/bigtop-deploy/puppet/modules/hadoop_zookeeper/templates/zoo.cfg
+index b646e60..e255a0a 100644
+--- a/bigtop-deploy/puppet/modules/hadoop_zookeeper/templates/zoo.cfg
++++ b/bigtop-deploy/puppet/modules/hadoop_zookeeper/templates/zoo.cfg
+@@ -26,8 +26,14 @@ syncLimit=5
+ dataDir=<%= @datadir %>
+ # the port at which the clients will connect
+ clientPort=<%= @port %>
+-<% @ensemble.each do |idx, ip| %>
++<% if !@ensemble[0].nil? && @ensemble[0].length == 2 %>
++  <% @ensemble.each do |idx, ip| %>
+ server.<%= idx %>=<%= ip %>
++  <% end %>
++<% else %>
++  <% @ensemble.each_with_index do |server,idx| %>
++server.<%= idx %>=<%= server %>
++  <% end %>
+ <% end %>
```

Comment:
I had github generate the patch this time, so it's all slightly different. Line 103-110 are the only differences from the main patch, though.